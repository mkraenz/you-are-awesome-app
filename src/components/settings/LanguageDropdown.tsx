import React, { FC, Fragment, useState } from "react";
import { useTranslation } from "react-i18next";
import { Divider, Menu } from "react-native-paper";
import { LanguageCodeToLocalizedLang } from "../../localization/localization";
import SettingsRow from "./SettingsRow";

const LanguageDropdown: FC = () => {
    const { t, i18n } = useTranslation();
    const languages = Object.values(LanguageCodeToLocalizedLang);
    const [visible, setVisible] = useState(false);

    const selectLanguage = (value: string) => {
        const selectedLangCode = Object.keys(
            LanguageCodeToLocalizedLang
        ).filter(
            (langCode) =>
                LanguageCodeToLocalizedLang[
                    langCode as keyof typeof LanguageCodeToLocalizedLang
                ] === value
        );
        if (selectedLangCode.length === 1) {
            i18n.changeLanguage(selectedLangCode[0]);
        }
    };
    return (
        <Menu
            visible={visible}
            onDismiss={() => setVisible(false)}
            anchor={
                <SettingsRow
                    onPress={() => setVisible(true)}
                    title={t("language")}
                ></SettingsRow>
            }
        >
            {languages.map((lang, i) => (
                <Fragment key={lang}>
                    <Menu.Item
                        key={lang}
                        onPress={() => {
                            selectLanguage(lang);
                            setVisible(false);
                        }}
                        title={lang}
                    />
                    {i !== languages.length - 1 && <Divider />}
                </Fragment>
            ))}
        </Menu>
    );
};

export default LanguageDropdown;
