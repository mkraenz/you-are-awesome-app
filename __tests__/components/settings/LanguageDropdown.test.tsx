import React from "react";
import "react-native";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import LanguageDropdown from "../../../src/components/settings/LanguageDropdown";
import TestLocalizationProvider from "../../helpers/TestLocalizationProvider";

it("renders correctly", () => {
    const tree = renderer
        .create(
            <TestLocalizationProvider>
                <LanguageDropdown />
            </TestLocalizationProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
