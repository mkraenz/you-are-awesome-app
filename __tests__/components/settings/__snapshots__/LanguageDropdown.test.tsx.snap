// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={false}
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginVertical": 6,
              "paddingLeft": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
                Object {},
              ],
            ]
          }
        >
          Language
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                  "letterSpacing": 0.5,
                  "lineHeight": 24,
                  "marginVertical": 2,
                },
                undefined,
              ],
            ],
          ]
        }
      >
        English
      </Text>
    </View>
  </View>
</View>
`;
